### PR TITLE
Use k8s.io/api/networking/v1 instead of k8s.io/api/extensions/v1beta1

### PR DESCRIPTION
**Pull request by towerhe**

### CODE REVIEW

**Comment by barnettZQG** [comment]
For goodrain/rainbond#927

File: controllers/handler/api.go [hunk 1]
```diff
@@ -422,12 +417,21 @@ func (a *api) ingressForWebsocket() client.Object {
...
-		Spec: extensions.IngressSpec{
-			Backend: &extensions.IngressBackend{
-				ServiceName: APIName + "-websocket",
-				ServicePort: intstr.FromString("ws"),
+		Spec: createIngressSpec(APIName+"-websocket", "ws", intstr.FromString("ws").IntVal),
+	}
+	return ing
+}
+
+func createIngressSpec(name, portName string, port int32) nwv1.IngressSpec {
```
Comment (barnettZQG) [inline controllers/handler/api.go:425]: Is it more semantic to change “name” to “serviceName”?

File: controllers/handler/app-ui.go [hunk 2]
```diff
@@ -294,10 +294,15 @@ func (a *appui) ingressForAppUI() client.Object {
 			},
 			Labels: a.labels,
 		},
-		Spec: extensions.IngressSpec{
-			Backend: &extensions.IngressBackend{
-				ServiceName: AppUIName,
-				ServicePort: intstr.FromString("http"),
+		Spec: nwv1.IngressSpec{
+			DefaultBackend: &nwv1.IngressBackend{
```
Comment (barnettZQG) [inline controllers/handler/app-ui.go:298]: use common function `createIngressSpec`?

**Review by barnettZQG — APPROVED** [review]
Ingress move to k8s.io/api/networking/v1 in v0.19.0.
we should be compatible with the api version before v0.19.0.

such as:
```
if kubeVersion.AtLeast(utilversion.MustParseSemantic("v0.19.0")) {
  // use k8s.io/api/networking/v1
}else{
  // use k8s.io/api/extensions/v1beta1
}
```

File: controllers/handler/api.go [hunk 3]
```diff
@@ -422,12 +417,21 @@ func (a *api) ingressForWebsocket() client.Object {
...
+				Name: serviceName,
+				Port: nwv1.ServiceBackendPort{
+					Name:   portName,
+					Number: port,
+				},
 			},
 		},
 	}
-	return ing
 }
```
Comment (towerhe) [inline controllers/handler/api.go:437]: Should I move this common function to `common.go`?

Comment (barnettZQG) [reply controllers/handler/api.go:437]: Yes.

File: controllers/handler/hub.go [hunk 4]
```diff
@@ -218,7 +218,7 @@ func (h *hub) persistentVolumeClaimForHub() *corev1.PersistentVolumeClaim {
 }
 
 func (h *hub) ingressForHub() client.Object {
-	ing := &extensions.Ingress{
+	ing := &nwv1.Ingress{
```
Comment (barnettZQG) [inline controllers/handler/hub.go:221]: Here, also need to be compatible with the API version.

**Comment by barnettZQG** [comment]
fiexed by #107

**Pull request closed without merging**